### PR TITLE
Gmail - Usability Audit

### LINKED ISSUE #13581: [Usability Audit] Gmail
In preparation of having Restricted Scopes provisioned on our Pipedream Gmail client, which will unlock our components to a much larger group of users, we should do a usability audit on all of the components. 

The easiest way to do this would most likely be to testing using Gmail (Developer App), and roll out changes on both Gmail (Developer App) as well as Gmail app.

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #13581 
@vunguyenhung Note to QA: Components will need to be tested using the Gmail Developer App.

## Summary by CodeRabbit

- **New Features**
	- Users can now add multiple labels to emails in the Gmail integration.
	- A new action allows for the removal of labels from emails.
	- Introduced a source for emitting events when new emails are received.
	- Enhanced capabilities for handling attachments in emails have been implemented.
	- New functionality to retrieve messages with attachments has been added.
	- A new method for managing labels on messages has been introduced.

- **Bug Fixes**
	- Corrected typographical errors in action names and improved error handling in attachment downloads.

- **Documentation**
	- Comprehensive setup guide for real-time email notifications has been added.
	- Updated version numbers across various components to reflect recent changes and enhancements.

### CODE REVIEW

File: components/gmail/sources/new-email-received-instant/new-email-received-instant.mjs [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+      hidden: true,
+    },
+  },
+  async additionalProps(props) {
+    const isValidClientId = await this.checkClientId();
+    props.appAlert.alertType = isValidClientId
+      ? "info"
+      : "error";
+    props.appAlert.content = isValidClientId
+      ? "OAuth client ID is valid. You can use the component."
```
Comment (malexanderlim) [inline components/gmail/sources/new-email-received-instant/new-email-received-instant.mjs:75]: I think we can omit the success case, and only throw a warning for the error case.

File: components/gmail/sources/new-email-received-instant/new-email-received-instant.mjs [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 24 * 60 * 60,
+      },
+      hidden: true,
+    },
+    db: "$.service.db",
+    topicType: {
+      type: "string",
+      label: "Type",
```
Comment (malexanderlim) [inline components/gmail/sources/new-email-received-instant/new-email-received-instant.mjs:30]: Could we change the name here to "Pub/Sub Topic"

**Review by luancazarine — DISMISSED** [review]
Hi @michelle0927, LGTM! Ready for QA!

**Review by luancazarine — APPROVED** [review]

**Review by malexanderlim — APPROVED** [review]

**Pull request merged**